src/Display/Action.py

```diff
@@ -0,0 +1,126 @@
+import tkinter as tk
+from random import random, choice
+
+from src.Data import Player
+from src.Util.Globals import *
+import src.Util.Database as db
+
+class PlayerActionScreen(tk.Frame):
+    instances = []
+    def __init__(self, team1_name, team2_name, team1_players: list[Player], team2_players: list[Player]):
+        master = tk.Tk()
+        super().__init__(master)
+        master.geometry('1200x800')
+        self.team1_name = team1_name
+        self.team2_name = team2_name
+
+        self.team1_score = 0
+        self.team2_score = 0
+
+        self.team1_players = team1_players
+        self.team2_players = team2_players
+
+        self.team1_player_scores: list[tk.StringVar] = []
+        for player in team1_players:
+            self.team1_player_scores.append(tk.StringVar(self, player.score))
+        self.team2_player_scores: list[tk.StringVar] = []
+        for player in team2_players:
+            self.team2_player_scores.append(tk.StringVar(self, player.score))
+
+        PlayerActionScreen.instances.append(self)
+
+        self.create_widgets()
+
+        self.time_left = 30
+        self.update_time_left()
+
+    def create_widgets(self):
+        self.team1_frame = tk.Frame(self)
+        self.team2_frame = tk.Frame(self)
+        self.countdown_frame = tk.Frame(self)
+
+        # Create labels for team names and scores
+        self.team1_label = tk.Label(self.team1_frame, text=self.team1_name, font=('Helvetica', 18), fg=red_color)
+        self.team1_label.grid(row=0, column=0)
+
+        self.team2_label = tk.Label(self.team2_frame, text=self.team2_name, font=('Helvetica', 18), fg=green_color)
+        self.team2_label.grid(row=0, column=2, pady=(20, 0))
+
+        self.team1_score_label = tk.Label(self.team1_frame, text=str(self.team1_score), font=('Helvetica', 25))
+        self.team1_score_label.grid(row=1, column=0)
+
+        self.team2_score_label = tk.Label(self.team2_frame, text=str(self.team2_score), font=('Helvetica', 25))
+        self.team2_score_label.grid(row=1, column=2)
+
+        self.team1_frame.grid(row=1, column=0, padx=200)
+        self.team2_frame.grid(row=1, column=2, padx=200)
+        self.countdown_frame.grid(row=0, column=0, columnspan=3)
+
+        # Create buttons for team actions
+        self.team1_button = tk.Button(self, text='Team 1 Action', command=self.team1_action)
+        self.team1_button.grid(row=2, column=0, padx=10, pady=10)
+
+        self.team2_button = tk.Button(self, text='Team 2 Action', command=self.team2_action)
+        self.team2_button.grid(row=2, column=2, padx=10, pady=10)
+
+        # Create label for time remaining
```

Comment:
just leaving a comment to block the PR - it isn't quite ready to be merged - Joe and I will work on this soon

Reply:
got it